src/cockpit/router.py

```diff
@@ -26,12 +27,54 @@
 logger = logging.getLogger(__name__)
 
 
+class ExecutionQueue(collections.deque):
```

Comment:
This is kinda something like the exact reason that I liked my previous approach a lot better.  Before, the queue object was just a queue.  Pure data.  Now it's this weird magical state tracking object glued to the side of a queue...

Reply:
It just seemed weird to me to have an API to wrap(), but no API to unwrap. I think I'd be okay with adding an explicit `unwrap` to hide that gross `del`. It still leaves the housekeeping to the API caller then, which is kind of ugly, but with a bunch of extra assertions this would be good enough, I figure.

Reply:
Also, I'd be fine with dismissing this alert, or adding a `NotImplementedError` stub.

Reply:
I took a different approach: ExecutionQueue is no longer a subclass of `deque`, but rather has it as an instance method.